### PR TITLE
Toolbar toggler accessibility

### LINKED ISSUE #37739: Fix accessibility of toolbar toggler
**Symfony version affected**: master

**Description**
Here is the currently used markup of "Show toolbar" link.
```html
<a href="#" title="Show Symfony toolbar" tabindex="-1" id="sfToolbarMiniToggler-95d68f" accesskey="D">
    <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24">
        <!-- SVG image -->
    </svg>
</a>
```

I think it has a few semantic and accessibility issues.

1. First of all, using `<a href="#"/>` here is not semantically correct, because it is not a link but button. It needs to be rendered with `<button/>` tag or at least have `role="button"`.
2. `tabindex=-1`. Why? It makes the toggler not reachable through keyboard. Note that `accesskey` cannot be considered as a replacement for `tabindex` because it may conflict with a system or browser keyboard shortcut. That's what is happening on my local machine. See [accessibility concerns](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/accesskey#Accessibility_concerns).
3. `aria-expanded` attribute is missing. Also we may consider adding `aria-controls` and `aria-haspopup` attributes.
4. SVG images need `aria-hidden=true` and `focusable=false` attributes.

### PR DESCRIPTION
**Pull request by Chi-teck**
| Q             | A
| ------------- | ---
| Branch       | master
| Bug fix      | yes
| New feature  | no
| Deprecations |no
| Tickets       | Fix #37739
| License       | MIT

### CODE REVIEW

**Comment by Chi-teck** [comment]
The PR does not combines toggler links into a single button as was suggested in
https://github.com/symfony/symfony/issues/37739#issuecomment-671898063 because it would require rewriting  a good amount of profiler CSS code which is out of scope of that ticket. Overall I think it makes sense though as the toggler would not loose focus when controlled via keyboard.

For the record, I believe the front-end part of the profiler needs deep refactoring. Using floats with clearfix hack in 2020 seems odd and fallbacks for `classList` and `EventListener` are not really needed nowadays. Dropping support for old IE versions would make CSS/JS part of the profiler a lot simpler and cleaner.

File: src/Symfony/Bundle/WebProfilerBundle/Resources/views/Profiler/toolbar.html.twig [hunk 1]
```diff
@@ -1,8 +1,8 @@
 <!-- START of Symfony Web Debug Toolbar -->
 <div id="sfMiniToolbar-{{ token }}" class="sf-minitoolbar" data-no-turbolink>
-    <a href="#" title="Show Symfony toolbar" tabindex="-1" id="sfToolbarMiniToggler-{{ token }}" accesskey="D">
+    <button class="show-botton" type="button" title="Show Symfony toolbar" id="sfToolbarMiniToggler-{{ token }}" accesskey="D" aria-expanded="false" aria-controls="sfToolbarMainContent-{{ token }}">
```
Comment (SHTIKOV) [inline src/Symfony/Bundle/WebProfilerBundle/Resources/views/Profiler/toolbar.html.twig:3]: Maybe did you mean `show-button` to `show-botton`?

Comment (Chi-teck) [reply src/Symfony/Bundle/WebProfilerBundle/Resources/views/Profiler/toolbar.html.twig]: That class is not really needed. I've just removed it.

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @Chi-teck.

**Comment by stof** [comment]
@Chi-teck the toolbar itself is intentionally keeping compat with old browsers, because it gets injected into the page. If it were broken in older browsers, it would forbid using Symfony to build apps supporting these older browsers (as you would not be able to test your page in those browsers locally due to the Symfony debug tools). The JS for the profiler pages is using newer features

**Pull request merged**